cmd/meroxa/flink/deploy.go

```diff
@@ -24,77 +26,105 @@ const (
 	irVal            = "EMIT_IR"
 )
 
+func doesJobUseMeroxaPlatform(ctx context.Context, jarPath string) (bool, error) {
+	cmd := exec.CommandContext(ctx, "jar", "-tf", jarPath)
+	output, err := cmd.Output()
+	if err != nil {
+		return false, err
+	}
+
+	scanner := bufio.NewScanner(bytes.NewBuffer(output))
+	scanner.Split(bufio.ScanLines)
+
+	for scanner.Scan() {
+		if strings.Contains(scanner.Text(), "com.meroxa") {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
 func GetIRSpec(ctx context.Context, jarPath string, secrets map[string]string, l log.Logger) (*ir.DeploymentSpec, error) {
 	if os.Getenv("UNIT_TEST") != "" {
 		return nil, nil
 	}
 
 	verifyJavaVersion(ctx, l)
 
-	cwd, err := os.Getwd()
+	usesMeroxa, err := doesJobUseMeroxaPlatform(ctx, jarPath)
 	if err != nil {
 		return nil, err
 	}
-	irFilepath := filepath.Join(cwd, irFilename)
-
-	cmd := exec.CommandContext(ctx, "java", "-jar", jarPath)
-	cmd.Env = append(
-		cmd.Environ(),
-		fmt.Sprintf("%s=%s", modeEnvVar, irVal),
-		fmt.Sprintf("%s=%s", outputEnvVar, irFilename))
-	output, err := cmd.CombinedOutput() // all java output goes to stderr, so that's fun
-	defer func() {
-		_ = os.Remove(irFilepath)
-	}()
-	if err != nil {
-		return nil, fmt.Errorf("%s\n%v", output, err)
-	}
 
-	_, err = os.Stat(irFilepath)
-	if err != nil {
-		// @TODO try the docker way because the jar is skinny
-		// Otherwise, there are no Meroxa* classes in this main class
-		return nil, nil
-	}
+	if usesMeroxa {
```

Comment:
thoughts on checking if `!usesMeroxa` and then returning early? that way the rest of this code is not nested in a condition?

We could also add a comment explaining why we only do `java -jar` when it has the bindings

Reply:
That's a good idea. Will definitely make it easier to read as well. 👏🏽